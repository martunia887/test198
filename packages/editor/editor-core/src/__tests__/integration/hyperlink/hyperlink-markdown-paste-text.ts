import { BrowserTestCase } from '@atlaskit/webdriver-runner/runner';
import Page from '@atlaskit/webdriver-runner/wd-wrapper';
import {
  expectMatchDocument,
  comment,
  fullpage,
  editable,
  copyToClipboard,
} from '../_helpers';
import {
  goToEditorTestingExample,
  mountEditor,
} from '../../__helpers/testing-example-helpers';

[comment, fullpage].forEach(editor => {
  BrowserTestCase(
    `hyperlink-markdown-paste-text.ts: Link - link markdown with pasting link text ${
      editor.name
    } editor`,
    {
      skip: ['ie', 'edge', 'safari'],
    },
    async (client: any, testName: string) => {
      const sample = new Page(client);
      await copyToClipboard(sample, 'https://hello.com');

      await goToEditorTestingExample(client, sample);
      await mountEditor(sample, { appearance: 'full-page' });

      await sample.type(editable, ['[link1](']);
      await sample.paste();
      await sample.type(editable, [')']);

      await expectMatchDocument(sample, testName);
    },
  );
});
