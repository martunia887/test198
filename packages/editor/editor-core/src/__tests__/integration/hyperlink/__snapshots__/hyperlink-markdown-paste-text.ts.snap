// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hyperlink-markdown-paste-text.ts: Link - link markdown with pasting link text comment editor 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "marks": Array [
            Object {
              "attrs": Object {
                "href": "https://hello.com",
              },
              "type": "link",
            },
          ],
          "text": "link1",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
  "version": 1,
}
`;

exports[`hyperlink-markdown-paste-text.ts: Link - link markdown with pasting link text fullpage editor 1`] = `
Object {
  "content": Array [
    Object {
      "content": Array [
        Object {
          "marks": Array [
            Object {
              "attrs": Object {
                "href": "https://hello.com",
              },
              "type": "link",
            },
          ],
          "text": "link1",
          "type": "text",
        },
      ],
      "type": "paragraph",
    },
  ],
  "type": "doc",
  "version": 1,
}
`;
